ts/client/src/accounts/mangoAccount.ts

```diff
@@ -1294,12 +1328,99 @@ export class PerpPosition {
   }
 
   public getPnl(perpMarket: PerpMarket): I80F48 {
+    if (perpMarket.perpMarketIndex !== this.marketIndex) {
+      throw new Error("PerpPosition doesn't belong to the given market!");
+    }
+
     return this.quotePositionNative.add(
       I80F48.fromI64(this.basePositionLots.mul(perpMarket.baseLotSize)).mul(
         perpMarket.price,
       ),
     );
   }
+
+  public updateSettleLimit(perpMarket: PerpMarket): void {
+    if (perpMarket.perpMarketIndex !== this.marketIndex) {
+      throw new Error("PerpPosition doesn't belong to the given market!");
+    }
+
+    const windowSize = perpMarket.settlePnlLimitWindowSizeTs;
+    const nowTs = new BN(Date.now() / 1000);
+    const newWindow = nowTs.gte(
```

Comment:
on dev, this compares the windowStart and windowEnd - maybe do that already or later when merging